File: Leetcode/1983/1983.cpp
class Solution {
public:
    int widestPairOfIndices(vector<int>& nums1, vector<int>& nums2) {
        int n = (int)nums1.size();
        vector<int> pre1(n+1);
        for (int i = 1; i < n+1; ++i) {
            pre1[i] = pre1[i-1]+nums1[i-1]; 
        }
        vector<int> pre2(n+1);
        for (int i = 1; i < n+1; ++i) {
            pre2[i] = pre2[i-1]+nums2[i-1];
        }
        unordered_map<int,vector<int>> unmap;
        for (int i = 0; i < n+1; ++i) {
            unmap[pre1[i]-pre2[i]].emplace_back(i);
        }
        int ans = 0;
        for (const auto &[k,v] : unmap) {
            if (v.size() > 1) {
                ans = max(v.back()-v.front(), ans);
            }
        }
        return ans;
    }
};

File: Leetcode/1983/1983.go
func widestPairOfIndices(nums1 []int, nums2 []int) int {
	pre1 := make([]int, len(nums1)+1)
	for i := 1; i < len(nums1)+1; i++ {
		pre1[i] = pre1[i-1] + nums1[i-1]
	}
	pre2 := make([]int, len(nums2)+1)
	for i := 1; i < len(nums2)+1; i++ {
		pre2[i] = pre2[i-1] + nums2[i-1]
	}
	unmap := make(map[int][]int)
	for i := 0; i < len(nums1)+1; i++ {
		unmap[pre1[i]-pre2[i]] = append(unmap[pre1[i]-pre2[i]], i)
	}
	ans := 0
	for _, v := range unmap {
		if len(v) > 1 {
			if ans < v[len(v)-1]-v[0] {
				ans = v[len(v)-1]-v[0]
			}
		}
	}
	return ans
}